pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -0,0 +1,562 @@
+	phase, _ := status["phase"]
+
+	switch phase {
+	case "creating":
+		// set provisioning to unknown
+		if !apimgmtv3.ClusterConditionProvisioned.IsUnknown(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionProvisioned.Unknown(cluster)
+			return e.clusterClient.Update(cluster)
+		}
+
+		// If there are no subnets it can be assumed that networking fields are not provided. In which case they
+		// should be created by the eks-operator, and needs to be copied to the cluster object.
+		if len(cluster.Spec.EKSConfig.Subnets) == 0 && cluster.Status.EKSStatus.GeneratedVirtualNetwork == "" {
+			if status["generatedVirtualNetwork"] != "" {
+				// network field have been generated and are ready to be copied
+				generatedVirtualNetwork, _ := status["generatedVirtualNetwork"].(string)
+				generatedSubnets, _ := status["generatedSubnets"].([]interface{})
+				generatedSecurityGroups, _ := status["generatedSecurityGroups"].([]interface{})
+				cluster = cluster.DeepCopy()
+				cluster.Status.EKSStatus.GeneratedVirtualNetwork = generatedVirtualNetwork
+				for _, val := range generatedSubnets {
+					cluster.Status.EKSStatus.GeneratedSubnets = append(cluster.Status.EKSStatus.GeneratedSubnets, val.(string))
+				}
+				for _, val := range generatedSecurityGroups {
+					cluster.Status.EKSStatus.GeneratedSecurityGroups = append(cluster.Status.EKSStatus.GeneratedSecurityGroups, val.(string))
+				}
+				return e.clusterClient.Update(cluster)
+			}
+		}
+		logrus.Infof("waiting for EKS cluster [%s] to finish creating", cluster.Name)
+		e.clusterEnqueueAfter(cluster.Name, 20*time.Second)
+		return cluster, nil
+	case "active":
+		if cluster.Spec.EKSConfig.Imported {
+			if cluster.Annotations == nil || cluster.Annotations[importedAnno] != "true" {
+				// imported annotation is not set- need to get spec from upstream cluster
+				var eksClusterConfigStructured eksv1.EKSClusterConfig
+				if err := runtime.DefaultUnstructuredConverter.FromUnstructured(eksClusterConfigDynamic.Object, &eksClusterConfigStructured); err != nil {
+					return cluster, fmt.Errorf("failed to import cluster [%s]", cluster.Name)
+				}
+
+				// copy EKS cluster config spec to cluster and set imported annotation
+				cluster = cluster.DeepCopy()
+				cluster.Spec.EKSConfig = &eksClusterConfigStructured.Spec
+				cluster.Status.AppliedSpec = cluster.Spec
+				if cluster.Annotations == nil {
+					cluster.Annotations = make(map[string]string)
+				}
+				cluster.Annotations[importedAnno] = "true"
+				cluster, err := e.clusterClient.Update(cluster)
+				if err != nil {
+					return cluster, err
+				}
+			}
+		}
+
+		if cluster.Spec.EKSConfig.Imported {
+			// cluster spec has been copied and cluster is no longer pending
+			if !apimgmtv3.ClusterConditionPending.IsTrue(cluster) {
+				cluster = cluster.DeepCopy()
+				apimgmtv3.ClusterConditionPending.True(cluster)
+				cluster, err = e.clusterClient.Update(cluster)
+			}
+		}
```

Comment:
Can this whole block move up under the same if check above then we can do just one update call to the cluster